### ISSUE TITLE
Flash Problems after ESP soldered on

### ISSUE THREAD

**Issue by JoeBerlin69** [codingcatgirl/split-flap-fae-pcb#3]
Hello, the boards made by JLPCB cannot be flashed if an ESP is soldered on. If both parts are separated, flashing works perfectly. When I uploaded the layout to JLPCB, there were problems with the arrangement of the components. I had to position many of them manually and, above all, rotate them. It is possible that I did something wrong here. Are you aware of similar problems? The units seem to work without ESP, so at least they rotate when power is applied and stop at the right place. Best regards

**Comment by codingcatgirl** [comment]
Just like the original PCB only has the ESP-01 on one PCB, you only need the ESP on one of these PCBs as well. 

You should have read the readme, which states so clearly in **bold text**.

> **Deselect assembly for the ESP2866 module, since you need it on one PCB only. You should solder it on yourself.**

JLCPCB not showing the components in the correct rotation in the preview is normal and can be ignored, as they also write next to the preview.

Maybe @evilscientress can say something about the flashing of the ESP, but such an issue would be news to me.

**Comment by JoeBerlin69** [comment]
Thank you for your quick reply. I also ordered and received the boards without the ESP. My problem is that after soldering on the ESP I can no longer establish a serial connection to either the ATmega or the ESP. I also don't know how to jumper the address bar. Perhaps you also know the problem with the serial connection.

**Comment by codingcatgirl** [comment]
The most likely explanation for this is simply that you didn't solder it on right. It can be a bit tricky, and if the entire board doesn't work that would indicate a short.

**Comment by scientress** [comment]
If both the ESP and the AVR don't respond anymore my guess is an issue with the RESET bus. Both MCUs have their RESET inputs connected. The line should have 3.3V.

**Comment by JoeBerlin69** [comment]
I will check it again. Funfact: i soldered 2 PCBs with an ESP. Both show exactly the same wrong behavior.
![20240707_194148.jpg](https://github.com/codingcatgirl/split-flap-fae-pcb/assets/99012149/529050d1-2908-4a30-aa80-26ae2e097ae1)

**Comment by codingcatgirl** [comment]
Maybe i'm not seeing it right, but this looks like you are using way too much solder, which can easily cause shorts somewhere. You should check using a multimeter.

**Comment by codingcatgirl** [comment]
Maybe the i'm not seeing it right, but this looks like you are using way too much solder, which can easily cause shorts somewhere. You should check using a multimeter.

**Comment by JoeBerlin69** [comment]
I tested it with the multimeter and couldn't find a short of neighboured pins. I also checked the connection of the serial pinout and the Pins. That seems fine. But I will next week check the parameters of my programmer. In the past i had some problems with them. The serial output was rubbish.
I can flash both chips without any problems if they're separated.
I will continue testing next week. Thanks a lot for your help.
Joerg

**Comment by jvdhoeven** [comment]
Maybe it helps: I was able to flash the atmega via the ISP pins using an arduino uno as programmer. I was unable to do it via the serial connections/pins on the bottom right, but did not investigate further. Flashing the ESP worked with no issue.

**Comment by scientress** [comment]
Yes you have to use ISP to burn the bootloader then you can use serial. That is normal. A AVR from the factory has no support for serial programming. You **have** to use ISP first and set the right fuses.

**Comment by JoeBerlin69** [comment]
As I told before, i hadn't a problem with the arduino single on board. The splitflap test routine worked well. I was able to  upload the unit ino and the offset ino and also calibrate the flaps. I also could flash the esp single with a programmer and run it with the test routines without the flaps. I got the split ap screen and could login in my wifi. After the that i soldered it on the pcb. Than i got the troubles. I also uncomment this line in the code. I made a mistake  with the unitcount so i had to reflash the ESP. That didn't work.
I also thought i made a bad soldering, so i gave me a second try. Unfortunately with exactly the same bad results.
Maybe s.o. lives in Germany and I can sent him an ESP and a PCB to test if the PCBs are ok.

**Comment by jonbloom** [comment]
I believe I flashed the bootloader via ISP succussfully, as now the board blinks green when applying power. Having said that, when connecting via the serial header (RX/TX/GND) toward the top of the board, I haven't had luck programming. I'm using a USB-TTL adapter on my Mac, which I have used elsewhere successfully. Basically just getting this error repeatedly:
```
avrdude: stk500_recv(): programmer is not responding
avrdude: stk500_getsync() attempt 1 of 10: not in sync: resp=0x00
```
